### ISSUE TITLE
Call of Duty: Black Ops 3 eventual XID 31 crash

### ISSUE THREAD

**Issue by BlueGoliath** [doitsujin/dxvk#2495]
With Proton GE 7.1.2(https://github.com/GloriousEggroll/proton-ge-custom/releases/tag/7.1-GE-2), Black Ops 3 will eventually crash. The crash happens about an hour usually while playing the Zombies mode even if the game is paused. 

The Nvidia driver spits out an XID 31 error, which is consistent on every crash after attempting to play about half a dozen games:

NVRM: Xid (PCI:0000:0c:00): 31, pid=11461, Ch 00000053, intr 10000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_2 faulted @ 0x0_00000000. Fault is of type FAULT_PDE ACCESS_TYPE_READ

System Info:

OS: Arch Linux(updated)
CPU: Ryzen 1800x
RAM: 32GB 3133Mhz 
GPU: Nvidia GeForce GTX 1080(Nvidia 510.47.03)

**Comment by Sporif** [comment]
Does this happen with the 495 driver series? I ask because I get that exact same error with the 510 series in certain dxvk games when switching focus from the game.

**Comment by BlueGoliath** [comment]
> Does this happen with the 495 driver series? I ask because I get that exact same error with the 510 series in certain dxvk games when switching focus from the game.

I don't remember having this issue on 495 but I don't exactly play games often anyway. It doesn't seem to have anything to do with window focus as it crashes while playing.

**Comment by doitsujin** [comment]
What does memory usage look like with `DXVK_HUD=memory`?

But yeah, as was said, we cannot debug or fix issues that take an hour to reproduce, especially when it's just another graphics driver crash.

**Comment by BlueGoliath** [comment]
The third heap was 96% full. Not sure if that matters or not.

I can reproduce this issue much faster with Civilization 6 with DX11. In addition to XID 31 I'm also getting XID 13:

NVRM: Xid (PCI:0000:0c:00): 13, pid=305, Graphics Exception:  MISSING_MACRO_DATA
NVRM: Xid (PCI:0000:0c:00): 13, pid=305, Graphics Exception: ESR 0x404490=0x80000001
NVRM: Xid (PCI:0000:0c:00): 13, pid=112370, Graphics Exception: ChID 0088, Class 0000c197, Offset 00002388, Data 30569a00

PID 305 is an Nvidia GPU driver process(IRQ 38).

**Comment by BlueGoliath** [comment]
Did that and it still crashes.

**Comment by BlueGoliath** [comment]
Yeah... it seems like there is something really wrong with the 510 series. Just got this:

NVRM: Xid (PCI:0000:0c:00): 69, pid=267621, Class Error: ChId 008e, Class 0000c197, Offset 00000a18, Data 80006420, ErrorCode 0000000c

When just standing around in BF4's target range map(or w/e you call it).

**Comment by Blisto91** [comment]
@BlueGoliath Friendly ping. Could i get you to try this again with latest dxvk (or latest proton GE or proton experimental) and newest driver.

**Comment by BlueGoliath** [comment]
I haven't been able to get a crash right away but that doesn't necessarily mean anything. I'll see if I can get it to happen with Civ6. Seems like Proton GE 7.30 broke backward compatibility or something because games do not like launching with it.

**Comment by BlueGoliath** [comment]
still happens.

`Xid (PCI:0000:0c:00): 31, pid=302112, name=BlackOps3.exe, Ch 00000063, intr 10000000. MMU Fault: ENGINE GRAPHICS GPCCLIENT_T1_1 faulted @ 0x0_00000000. Fault is of type FAULT_PDE ACCESS_TYPE_READ`

**Comment by Blisto91** [comment]
The new 535 driver have alot of fixes, might be worth a try.

**Comment by Blisto91** [comment]
Aw sorry to hear about your 1080. RIP

**Comment by Blisto91** [comment]
@BlueGoliath I assume you haven't had any issues since?

**Comment by Blisto91** [comment]
@BlueGoliath Friendly ping

**Comment by BlueGoliath** [comment]
I didn't even see the comment in the notifications. Why close it now?

**Comment by Blisto91** [comment]
Old issue without further feedback that we weren't aware it was still a problem.
Have you hit the issue since?

**Comment by BlueGoliath** [comment]
I don't play it. Nvidia's driver is broke and you removed the option to limit VRAM supposedly so it's unplayable.

**Comment by Blisto91** [comment]
If you mean `dxgi.maxDeviceMemory` then it is still around, but isn't a hard cap and never was. Just tells the game you have less memory than you have, but the game can choose to ignore it.

The issue which caused a driver crash in the CoD d3d11 remasters is gone and so i hope this is too.

**Comment by BlueGoliath** [comment]
I remember reading a post on Reddit saying that you did but maybe i'm misremembering.

Anyway, if you feel like it's fixed then close it. I don't really care.

Edit: forgot I'm the author. Guess i'll just close it.
